pkg/config/sysprep.go

```diff
@@ -66,31 +66,41 @@ func validateAutounattendPresence(dirPath string) error {
 // CreateSysprepDisks creates Sysprep iso disks which are attached to vmis from either ConfigMap or Secret as a source
 func CreateSysprepDisks(vmi *v1.VirtualMachineInstance) error {
 	for _, volume := range vmi.Spec.Volumes {
-		if volume.Sysprep != nil {
-			if sysprepVolumeHasContents(*volume.Sysprep) {
-				var filesPath []string
+		if shouldCreateSysprepDisk(volume) {
```

Comment:
Now you can reduce complexity even further:
```suggestion
		if !shouldCreateSysprepDisk(volume) {
		  return nil
		}
		if err := createSysprepDisk(volume); err != nil {
			return err
		}
		
```

Reply:
Hm, this may not work IIUC. We loop over all VM volumes, so for example, if there are two disks and the sysprep one happens to be the second, than with this change we would return nil and never created the sysprep disk, right?

Reply:
Okay, then `continue` instead

Reply:
Your suggestion makes sense and with `continue` it'll definitely work, but is it gonna improve the code? The cognitive complexity would go down, but imo readability of the code will be worse. That's why I'd rather keep this condition unchanged.

Reply:
The complexity rule exists for readability.
Exiting early is a common practice especially in Golang and should be used where possible to reduce indentation where possible, imo.

Reply:
A good source on some proverbs for Go (but honestly any other language): https://dave.cheney.net/2020/02/23/the-zen-of-go